### DISCUSSION TITLE
Error when Listing an ERC-721 and fulfilling it

### DISCUSSION THREAD

**Discussion by Harvey44** [ProjectOpenSea/seaport#564] [Q&A] [answered]
Trying to fulfill an order with these, but been unable to, so far

const offerer = "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266";
const fulfiller = "0x0BB5690957d9def9Ac09dA2e9666cb15a4D43800";
const { executeAllActions } = await seaport.createOrder({
  offer: [
    {
      itemType: ItemType.ERC721,
      token: "0x8a90cab2b38dba80c64b7734e58ee1db38b8992e",
      identifier: "1",
    }
  ],
  consideration: [
    {
      amount: parseEther("0.4").toString(),
      recipient: offerer,
    }
  ],
  accountAddress: offerer
});

const order = await executeAllActions();

const { executeAllActions: executeAllFulfillActions } = await seaport.fulfillOrder({
  order,
  accountAddress: fulfiller,
});

const transaction = executeAllFulfillActions()

When i executed this, i got an approval request and Eip-712 payload, but then this error comes up 

Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0x0BB5690957d9def9Ac09dA2e9666cb15a4D43800","to":"0x00000000006c3852cbEf3e08E8dF289169EdE581","value":{"type":"BigNumber","hex":"0x51dac207a000"},,"accessList":null}, error={"code":-32603,"message":"execution reverted","data":{"originalError":{"code":3,"data":"0x6f7eac26","message":"execution reverted"}}}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.6.8)

what am i missing pls

**Comment by arr00** [comment]
This question is related to [seaport-js](https://github.com/ProjectOpenSea/seaport-js), not seaport; please ask the question on the seaport-js repository.

**Comment by JasperAlexander** [comment, marked as answer]
Try this code.

```
const sell = async() => {
    if(typeof seaport !== 'undefined') {
      const { executeAllActions } = await seaport.createOrder({
        offer: [
          {
            itemType: ItemType.ERC721,
            token: testERC721Address, // The ERC721 token address
            identifier: "1"
          }
        ],
        consideration: [
          {
            amount: parseEther("10").toString(),
            token: testERC20Address, // The ERC20 token address
            recipient: offerer // Account address of offerer
          }
        ]
      })

      const executedOrder = await executeAllActions()
      setCurrentOrder(executedOrder)  // Save to state
    }
  }

  const buy = async() => {
    if(typeof seaport !== 'undefined' && currentOrder) {
      const { executeAllActions: executeAllFulfillActions } = await seaport.fulfillOrder({
        order: currentOrder, // Retrieve from state
        accountAddress: fulfiller // Account address of fulfiller
      })

      const transaction = await executeAllFulfillActions()
    }
  }
```

**Reply by Harvey44** [reply]
am getting same error, is there some action i should be doing with the fulfiller address first ?

**Reply by JasperAlexander** [reply]
> am getting same error, is there some action i should be doing with the fulfiller address first ?

Can you list the actions you're doing which leads to this error?

**Reply by Harvey44** [reply]
connected to the offerer wallet, sign approval to seaport , approved signature request, then error

**Reply by JasperAlexander** [reply]
> connected to the offerer wallet, sign approval to seaport , approved signature request, then error

Have you minted the right tokens? Try reconnecting to the network in MetaMask and reset your MetaMask account (transactions).
